packages/react/src/components/button/button.react.stories.tsx

```diff
@@ -27,7 +27,17 @@ export default {
   },
 } as Meta<ButtonProps>;
 
-export const Playground: Story<ButtonProps> = (props) => <Button {...props} />;
+export const Playground: Story<ButtonProps> = ({
+  href,
+  ...restProps
+}: ButtonProps & Pick<JSX.IntrinsicElements['a'], 'href'>) => (
+  <Button
+    {...{
+      ...restProps,
+      ...(href && { href }), // don't set "href" key when value is empty string
+    }}
```

Comment:
I don't like using property spreads like this mainly for two reasons:
- TypeScript can't properly pick up the keys and allows things it shouldn't
- When one of the properties' types err, the message is really confusing and not helpful as it should be

e.g.
![image](https://user-images.githubusercontent.com/18623773/104300938-fc661500-54be-11eb-929d-6d3c04b69de6.png)

You can see it doesn't complain about the spread `href`, but does so correctly in the line below that.

I'd really prefer this: (a bit terser too)
```suggestion
    {...restProps}
    href={href || undefined} // don't set "href" when value is falsy
```

Oh and the comment change is a nitpick suggestion.

Reply:
Unfortunately this does not work...

Storybook in case of `undefined` argument does not forward it to props:

<img width="210" alt="Screenshot 2021-01-12 at 12 53 33" src="https://user-images.githubusercontent.com/20243687/104305299-6c2ace80-54c4-11eb-928f-50f07dcd9fc7.png">

But with this change `href` would always be there even `undefined`, and Button component explicitly checks for prop to be present (not for whether it is defined). This is because of how component overloads are designers.

Reply:
😭  🤷🏻